Push/PushRegistration.swift

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public class PushRegistration: NSObject, NSCoding {
+    let endpoint: NSURL
+    let uaid: String
+    let secret: String
+    let channelID: String
+
+    public init(uaid: String, secret: String, endpoint: NSURL, channelID: String) {
+        self.uaid = uaid
+        self.secret = secret
+        self.endpoint = endpoint
+        self.channelID = channelID
+    }
+
+    @objc public convenience required init?(coder aDecoder: NSCoder) {
+        guard let uaid = aDecoder.decodeObjectForKey("uaid") as? String,
+            let secret = aDecoder.decodeObjectForKey("secret") as? String,
+            let urlString = aDecoder.decodeObjectForKey("endpoint") as? String,
+            let endpoint = NSURL(string: urlString),
+            let channelID = aDecoder.decodeObjectForKey("channelID") as? String else {
+                fatalError("Cannot decode registration")
```

Comment:
Since this initializer is failable returning nil here might make more sense as I don't see decoding issues being a fatal error or developer specific error.